src/startupscriptgenerator/src/dotnetcore/scriptgenerator.go

```diff
@@ -42,6 +42,16 @@ func (gen *DotnetCoreStartupScriptGenerator) GenerateEntrypointScript(scriptBuil
 	}
 
 	dotnetBinary := "/usr/share/dotnet/dotnet"
+
+	preRunScriptPath := common.GetPathEnvironmentVariable(consts.AppServicePreRunScriptFilePathEnvVarName)
+	if preRunScriptPath != "" && !common.PathExists(dotnetBinary) {
```

Comment:
As mentioned above, this variable could be a list of commands or a relative path to a script file.

Reply:
Sounds good. I'm leaning to either a relative path to a script file or a single string command. A list of commands might be redundant here?